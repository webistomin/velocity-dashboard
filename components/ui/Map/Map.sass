+b('map')
  +e('map')
    &.vue2leaflet-map
      height: 70vh
      max-height: rem(700px)
      margin-bottom: rem(24px)

  +e('passenger-heading')
    display: flex
    margin-bottom: rem(24px)

  +e('passenger-avatar')
    margin-right: rem(10px)

  +e('passenger-info')
    display: flex
    flex-direction: column

  +e('passenger-name')
    font-weight: 500
    margin-bottom: rem(4px)

  +e('passenger-count')
    color: var(--color-gray-3)

  +e('passenger-item')
    display: flex
    flex-direction: column

    &:not(:last-of-type)
      margin-bottom: rem(16px)

  +e('passenger-caption')
    margin-bottom: rem(10px)

  +e('passenger-payment')
    display: flex
    align-items: center
    justify-content: center
    padding: rem(24px)
    border-radius: 5px
    border: rem(1px) solid var(--color-lavender-2)
    max-width: rem(140px)

  +e('passenger-payment-logo')
    width: 50%

  +e('passenger-col')
    +m('left')
      margin-bottom: rem(24px)

      +media($screen-sm)
        margin-bottom: 0
        flex-shrink: 0
        margin-right: rem(40px)

    +m('right')
      +media($screen-sm)
        width: 100%

  +e('passenger-payments')
    display: grid
    grid-gap: rem(24px)
    grid-template-columns: repeat(auto-fill, minmax(140px, 1fr))

  +e('passenger')
    +media($screen-sm)
      display: flex

  +e('grid')
    display: grid
    justify-content: start
    grid-template-columns: repeat(12, 1fr)
    grid-gap: rem(14px)

    +media($screen-sm)
      grid-gap: rem(24px)

    +media($screen-lg)
      grid-gap: rem(34px)

  +e('block')
    &:first-of-type
      grid-column: 12 span
      grid-row: 2 span

    &:nth-of-type(2)
      grid-column: 12 span
      grid-row: 1 span

    &:nth-of-type(3)
      grid-column: 12 span
      grid-row: 1 span

    +media($screen-lg)
      &:first-of-type
        grid-column: 7 span
        grid-row: 4 span

      &:nth-of-type(2)
        grid-column: 5 span
        grid-row: 3 span

      &:nth-of-type(3)
        grid-column: 5 span
        grid-row: 1 span


