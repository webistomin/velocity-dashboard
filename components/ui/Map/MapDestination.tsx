import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { BaseIcon } from 'components/base/BaseIcon/BaseIcon';
import { VNode } from '~/node_modules/vue';

@Component({
  name: 'MapDestination',
})
export default class MapDestination extends VueComponent {
  public render(): VNode {
    return (
      <div class='map__destination'>
        <div class='map__destination-heading'>
          <div class='map__destination-point'>
            <BaseIcon class='map__destination-icon' name='icon-check' color='blue' size='xs' />
            <strong class='map__destination-name'>Jackson Heights</strong>
            <span class='map__destination-street'>37-27 74th Street</span>
          </div>
          <div class='map__destination-point'>
            <BaseIcon class='map__destination-icon' name='icon-pin' color='green' size='xs' />
            <strong class='map__destination-name'>Greenpoint</strong>
            <span class='map__destination-street'>81 Gate St Brooklyn</span>
          </div>
        </div>
        <div class='map__destination-content'>
          <ul class='map__destination-info list'>
            <li class='map__destination-item list-item'>
              <strong class='map__destination-value'>12.3 km</strong>
              <span class='map__destination-key'>Distance</span>
            </li>
            <li class='map__destination-item list-item'>
              <strong class='map__destination-value'>42 min</strong>
              <span class='map__destination-key'>Time</span>
            </li>
            <li class='map__destination-item list-item'>
              <strong class='map__destination-value'>$34.20</strong>
              <span class='map__destination-key'>Price</span>
            </li>
            <li class='map__destination-item list-item'>
              <strong class='map__destination-value'>12.4 kWh</strong>
              <span class='map__destination-key'>Energy</span>
            </li>
          </ul>
        </div>
      </div>
    );
  }
}
