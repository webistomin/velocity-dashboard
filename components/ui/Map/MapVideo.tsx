import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

@Component({
  name: 'MapVideo',
})
export default class MapVideo extends VueComponent {
  render(): VNode {
    return (
      <div class='map__video'>
        <button class='map__video-play-btn btn'>
          <img class='map__video-play-image image' src='/img/interior.jpg' alt='Video' />
          <svg-icon name='icon-play' width={18} height={20} />
        </button>
        <div class='map__video-content'>
          <div class='map__video-info'>
            <strong class='map__video-name'>Activate interior video</strong>
            <span class='map__video-place'>37-27 74th Street</span>
          </div>
          <img class='map__video-camera image' src='/img/camera.svg' width={24} height={24} alt='Camera' />
        </div>
      </div>
    );
  }
}
