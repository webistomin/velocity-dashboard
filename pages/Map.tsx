import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import Map from 'components/ui/Map';

@Component({
  name: 'MapPage',
  middleware: ['auth-guard', 'auth-admin'],
})
export default class MapPage extends VueComponent {
  render(): VNode {
    return (
      <main class='page-content page-content_offset-bottom'>
        <Map />
      </main>
    );
  }
}
