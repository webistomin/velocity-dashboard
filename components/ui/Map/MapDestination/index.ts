import { MapDestination } from './MapDestination';

export default MapDestination;
