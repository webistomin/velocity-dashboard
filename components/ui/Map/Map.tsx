import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import BaseBlock from 'components/base/BaseBlock';
import BaseMap from 'components/base/BaseMap';
import MapPassenger from './MapPassenger';
import MapDestination from './MapDestination';
import MapVideo from './MapVideo';

import './Map.sass';

@Component({
  name: 'Map',
})
export default class Map extends VueComponent {
  public render(): VNode {
    return (
      <section class='map'>
        <BaseMap class='map__map' />
        <div class='map__container container'>
          <div class='map__grid'>
            <BaseBlock class='map__block' title='Passenger info'>
              <MapPassenger />
            </BaseBlock>
            <BaseBlock class='map__block'>
              <MapDestination />
            </BaseBlock>
            <BaseBlock class='map__block'>
              <MapVideo />
            </BaseBlock>
          </div>
        </div>
      </section>
    );
  }
}
