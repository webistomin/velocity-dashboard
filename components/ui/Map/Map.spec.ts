import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Map from './Map';

describe('Map', () => {
  const localVue = createLocalVue();
  localVue.use(VueRouter);
  const router = new VueRouter();

  const factory = (propsData = {}, options = {}) => {
    return shallowMount(Map, {
      localVue,
      router,
      propsData: {
        isNavOpened: false,
        ...propsData,
      },
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });
});
