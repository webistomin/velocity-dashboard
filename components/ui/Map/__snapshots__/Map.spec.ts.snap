// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Matches snapshot 1`] = `
"<section class=\\"map\\">
  <div class=\\"map__container container\\">
    <baseblock-stub title=\\"Hint\\" class=\\"map__placeholder\\">
      <vuecomponent-stub level=\\"3\\">Information about trip will be available after click on marker</vuecomponent-stub>
    </baseblock-stub>
  </div>
  <basemodal-stub transitionname=\\"modal-fade\\"><video preload=\\"auto\\" playsinline=\\"true\\" loop=\\"loop\\" class=\\"map__driver-video\\">
      <source src=\\"/video/car-driving.webm\\" type=\\"video/webm; codecs=&quot;vp8, vorbis&quot;\\">
      <source src=\\"/video/car-driving.mp4\\" type=\\"video/mp4; codecs=&quot;avc1.42E01E, mp4a.40.2&quot;\\">Video tag is not supported in this browser.</video></basemodal-stub>
</section>"
`;
